src/compiler/emitter.ts

```diff
@@ -3378,8 +3379,8 @@ namespace ts {
         }
 
         function emitJsxExpression(node: JsxExpression) {
-            if (node.expression) {
-                writePunctuation("{");
+            if (node.expression || (!commentsDisabled && !nodeIsSynthesized(node) && getTextOfNode(node).length > 2 && getTextOfNode(node).indexOf("/*") > -1)) { // preserve empty expressions if they contain comments!

```

Comment:
`hasCommentAtPosition` isn't a thing we currently have, but this does a, _okay_  job of approximating it. Obviously doesn't handle something like
```ts
const x = <div>
{ // first line 
/*last line*/}
</div>;
```
Maybe it's worth doing a full check with `forEachTrailingCommentRange`? @rbuckton - your opinion?

Reply:
do we care about `//` comments? The new check only handles `/**/` comments, right?

Reply:
Yep. That's why I'm asking if it's worth doing a full comment range iteration to check if there are any comments attached to the position.